alf/algorithms/agent.py

```diff
@@ -393,6 +395,12 @@ def after_train_iter(self, experience, train_info: AgentInfo = None):
         algorithms = list(filter(lambda a: a is not None, algorithms))
         self._agent_helper.after_train_iter(algorithms, experience, train_info)
 
+        # Strictly speaking, if we load a checkpoint for training, then in the
+        # first iteration ``self._rl_algorithm`` might have the wrong reward
+        # weights because the function call below is called only after the first
+        # iteration. But for simplicity (there is no easy way to set the weights
+        # after loading a checkpoint while before the first iteration) here we
+        # ignore its effect.
```

Comment:
All the changes to parameters will be overridden by checkpoint.  See Trainer._restore_checkpoint(). But there will be some experiences written to replay buffer (if replay buffer is not checkpointed).

Reply:
> All the changes to parameters will be overridden by checkpoint. See Trainer._restore_checkpoint(). But there will be some experiences written to replay buffer (if replay buffer is not checkpointed).

After loading a checkpoint, the first thing is that self._reward_weight_algorithms.reward_weights is updated. Then we need to call set_reward_weights on self._rl_algorithm (otherwise it's all ones by default and it is not a parameter in RLAlgorithm).  Even though there are some experiences written to replay buffer in the first iteration (generated by wrong reward weights), I think it's generally OK for an off-policy algorithm.